synapse/cmds/cron.py

```diff
@@ -0,0 +1,598 @@
+
+    async def runCmdOpts(self, opts):
+        line = opts.get('line')
+        if line is None:
+            self.printf(self.__doc__)
+            return
+
+        core = self.getCmdItem()
+
+        parseinfo = await s_syntax.getRemoteParseInfo(core)
+        argv = s_syntax.Parser(parseinfo, line).stormcmd()
+        try:
+            opts = self._make_argparser().parse_args(argv)
+        except s_exc.ParserExit:
+            return
+
+        handlers = {
+            'add': self._handle_add,
+            'list': self._handle_list,
+            'mod': self._handle_mod,
+            'del': self._handle_del,
+            'stat': self._handle_stat
+        }
+        await handlers[opts.cmd](core, opts)
+
+class At(s_cli.Cmd):
+    '''
+Adds a non-recurring cron job that will execute a Storm query at one or more
+specified times.
+
+List/details/deleting cron jobs created with 'at' use the same commands as
+other cron jobs:  cron list/stat/del respectively.
+
+Syntax: at (time|+time delta)+ {query}
+
+Notes: This command accepts one or more time specifications followed by exactly
+one storm query in curly braces.  Each time specification may be in synapse
+time delta format (e.g + 1 day) or synapse time format (e.g.
+20501217030432101).  Seconds will be ignored, as cron jobs' granularity is
+limited to minutes.
+
+All times are interpreted as UTC.
+
+The other option for time specification is a relative time from now.  This
+consists of a plus sign, a positive integer, then one of 'minutes, hours,
+days'.
+
+Note that the record for a cron job is stored until explicitly deleted via
+"cron del".
+
+Examples:
+    # Run a storm query in 5 minutes
+    at +5 minutes {[inet:ipv4=1]}
+
+    # Run a storm query tomorrow and in a week
+    at +1 day +7 days {[inet:ipv4=1]}
+
+    # Run a query at the end of the year Zulu
+    at 20181231Z2359 15:30 {[inet:ipv4=1]}
+'''
+    _cmd_name = 'at'
+
+    _cmd_syntax = (
+        ('line', {'type': 'glob'}),
+    )
+
+    def _make_argparser(self):
+        parser = s_cmd.Parser(prog='at', outp=self, description=self.__doc__)
+        parser.add_argument('args', nargs='+', help='date | delta| {query})')
+        return parser
+
+    async def runCmdOpts(self, opts):
+        line = opts.get('line')
+        if line is None:
+            self.printf(self.__doc__)
+            return
+
+        core = self.getCmdItem()
+
+        parseinfo = await s_syntax.getRemoteParseInfo(core)
+        argv = s_syntax.Parser(parseinfo, line).stormcmd()
+        # Currently, using an argparser is overkill for this command.  Using for future extensibility (and help).
+        try:
+            opts = self._make_argparser().parse_args(argv)
+        except s_exc.ParserExit:
+            return
+
+        query = None
+        consumed_next = False
+
+        tslist = []
+        # TODO: retrieve time from cortex in case of wrong cmdr time
+        now = time.time()
```

Comment:
Do we just want to plumb this capability in now while were in this PR? Alternatively, since we already have Locale issues potentially in terms of date parsing for cron jobs; push ALL of the parsing to the Cortex? In our prebuilt docker containers we can ensure that Locale is set properly.

Reply:
No locale-related parsing occurs in cortex. Let's keep it to the client where it belongs.
Made a task for the time getting.

Reply:
I purposefully converted anything locale-specific in the cmdr. There is no locale-specific parsing in the cortex. Luckily we all agree on the Arabic numeral system, and 0==Monday is a ISO 8601 thing.